File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        freq = Counter(s)
        a1,a2=0,100
        
        for val in freq.values():
            if val&1:
                a1=max(a1,val)
            else:
                a2=min(a2,val)
        return a1-a2

File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        int a1=0, a2=100;

        unordered_map<char,int> um;

        for(char ch: s)um[ch]++;

        for(auto it: um)
        {
            if(it.second&1)
                a1=max(a1,it.second);
            else
                a2=min(a2,it.second);
        }

        return a1-a2;
        
    }
};